lib/libc/minimal/source/string/string.c

```diff
@@ -203,7 +203,7 @@ void *memmove(void *d, const void *s, size_t n)
 	char *dest = d;
 	const char *src  = s;
 
-	if ((size_t) (d - s) < n) {
+	if ((size_t) (dest - src) < n) {
```

Comment:
ok, we may have a problem here, shouldn't you cast to `ssize_t` to avoid an underflow ? Instead of using `char *` it may be better using something `uint8_t` from `stdint.h` that is part of C standard and is recommended by other code guidelines.

Reply:
> ok, we may have a problem here, shouldn't you cast to ssize_t to avoid an underflow ? 

Nope. I tried to make an ASCII drawing of why size_t is correct:
```
A):
    |            ssssssssssss  
    |      dddddddddddd         must be written forwards

                if ((size_t) (0 - 10) < 12) {
                if ((size_t) (-10) < 12) {
                if ((size_t) (-10) < 12) {
                if (4294967286 < 12) {      or some other big number
                if (false) {    False, so we do forward writes

B):
    |ssssssssssss              
    |      dddddddddddd         must be written backwards
 
                if ((size_t) (dest - src) < n) {
                if ((size_t) (10 - 0) < 12) {
                if ((size_t) 10 < 12) {
                if (10 < 12) {
                if (true) {     True, so we do backwards writes

C):
    |      dddddddddddd        dont care, could be no-op
    |      ssssssssssss
```

Reply:
> Instead of using char * it may be better using something uint8_t from stdint.h that is part of C standard and is recommended by other code guidelines.

I'm confortable with char*, but agree uint8_t* would be better. However I also know of  platforms having problems with stdint.h.